processing/src/main/java/io/druid/query/aggregation/FilteredAggregatorFactory.java

```diff
@@ -112,7 +127,11 @@ public Object finalizeComputation(Object object)
   @Override
   public String getName()
   {
-    return delegate.getName();
+    String name = delegate.getName();
+    if (Strings.isNullOrEmpty(name)) {
+      name = this.name;
+    }
+    return name;
```

Comment:
It would be wierd to ignore the name set in FilteredAggregator when user explicitly sets it.
 
IMO, the default should be - If user has set a name for FilteredAggregatorFactory - use that, otherwise use the name of the delegate. For backwards compatible case user would not have specified name at filtered aggregator level and will get the old behavior of getting delegate name.

Reply:
seems like unintentional because this defeats the purpose in PR description :)

Reply:
@nishantmonu51 fixed